plugins/node/opentelemetry-instrumentation-ioredis/src/utils.ts

```diff
@@ -17,6 +17,47 @@
 import { Span, SpanStatusCode } from '@opentelemetry/api';
 import { DbStatementSerializer } from './types';
 
+/**
+ * Map of command names to the number of arguments that should be serialized.
+ * For example, SET should serialize which key it's operating on, but not its value.
+ * Commands not listed will have all their arguments serialized.
```

Comment:
Ideally, I think we would probably want to list which commands should capture all arguments, and then fall back to collecting just command name for commands not in the white list.

This will cover things like [define custom commands](https://github.com/luin/ioredis#lua-scripting) and will generally be more defensive and secure.

Since I guess writing such a list could be very long and not practical to write, I am good with keeping it as is. Since you did some research, I am curious if you have any thoughts on this

Reply:
> Ideally, I think we would probably want to list which commands should capture all arguments, and then fall back to collecting just command name for commands not in the white list.

I think there's definite value in capturing some arguments, even when we're not capturing all. Especially for SET/HSET/LPUSH and the like, we really want to register which keys were modified. For pub/sub, we want to know all the channels that were subscribed to, and so on.

> This will cover things like [define custom commands](https://github.com/luin/ioredis#lua-scripting) and will generally be more defensive and secure.

Perhaps a config option for additional commands to obfuscate would solve the same problem? One consequence of making the default serializer more opinionated is that there's more to replace if you decide to roll your own. This may have gotten to the point where it should be possible to append to the default as well as replace it entirely.